### PR TITLE
added simple operations and clear button funcionality

### PR DESCRIPTION
**Pull request by DiegoDiazGlobant**
Now the calculator does basic operations.
Implemented the clear button (last value and all values).

Preview:

https://user-images.githubusercontent.com/75440409/105360400-4c3b8f00-5bd7-11eb-8513-941c8908d913.mp4

### CODE REVIEW

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 1]
```diff
@@ -23,25 +23,42 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        setUpClickListenerToNumberButton(binding.buttonFive);
+        setUpClickListenerToNumberButton(binding.buttonSix);
+        setUpClickListenerToNumberButton(binding.buttonSeven);
+        setUpClickListenerToNumberButton(binding.buttonEight);
+        setUpClickListenerToNumberButton(binding.buttonNine);
     }
 
-    private void setUpClickListenerToButton(Button buttonPressed) {
-        buttonPressed.setOnClickListener(view -> presenter.onButtonPressed(buttonPressed.getText().toString()));
+    private void setUpClickListenerToNumberButton(Button buttonPressed) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/calculator/MainActivity.java:45]: if this method is only called one time, I think is not necessary. You can call directly in the listener

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 2]
```diff
@@ -23,25 +23,42 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+    private void setUpClickListenerToOperatorButton(Button buttonPressed) {
+        buttonPressed.setOnClickListener(view -> presenter.onOperatorButtonPressed(buttonPressed.getText().toString()));
+    }
+
+    private void setUpClickListenerToEqualsButton(Button buttonPressed) {
+        buttonPressed.setOnClickListener(view -> presenter.onEqualsButtonPressed());
+    }
+
+    private void setUpClickListenerToClearLastButton(Button buttonPressed) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/calculator/MainActivity.java]: same here

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 3]
```diff
@@ -23,25 +23,42 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        buttonPressed.setOnClickListener(view -> presenter.onButtonPressed(buttonPressed.getText().toString()));
+    private void setUpClickListenerToNumberButton(Button buttonPressed) {
+        buttonPressed.setOnClickListener(view -> presenter.onNumberButtonPressed(buttonPressed.getText().toString()));
+    }
+
+    private void setUpClickListenerToOperatorButton(Button buttonPressed) {
+        buttonPressed.setOnClickListener(view -> presenter.onOperatorButtonPressed(buttonPressed.getText().toString()));
+    }
+
+    private void setUpClickListenerToEqualsButton(Button buttonPressed) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/calculator/MainActivity.java]: same here

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java [hunk 4]
```diff
@@ -1,18 +1,95 @@
...
+    }
+
+    public boolean incompleteOperation() {
+        return (!operator.isEmpty() && secondOperand.isEmpty());
+    }
+
+    @Override
+    public void deleteLast() {
+        if (!secondOperand.isEmpty()) {
+            secondOperand = secondOperand.substring(0, secondOperand.length() - 1);
```
Comment (BQuinteros) [inline app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: pls remove this hardcode, move to a constant

File: README.md [hunk 5]
```diff
@@ -2,4 +2,12 @@
 
 #### 1/15/2021
 
-##### diegodiaz/calculator_ui
\ No newline at end of file
+##### diegodiaz/calculator_ui
```
Comment (SartiMau) [inline README.md:5]: please add a brief explanation about what happened on this commit

File: app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java [hunk 6]
```diff
@@ -17,4 +21,26 @@ public CalculatorView(Activity activity, ActivityMainBinding binding) {
...
     }
+
+    @Override
+    public void showResultValue(String result) {
+        binding.textViewResult.setText(result);
+    }
+
+    @Override
+    public void showWrongOperator() {
+        showResultValue(WRONG_OPERATOR_ERROR);
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java]: do not use constants here. you are in the view, so you can access string resources

File: app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java [hunk 7]
```diff
@@ -17,4 +21,26 @@ public CalculatorView(Activity activity, ActivityMainBinding binding) {
...
+    }
+
+    @Override
+    public void clearValues() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java:33]: same on this method

File: app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java [hunk 8]
```diff
@@ -17,4 +21,26 @@ public CalculatorView(Activity activity, ActivityMainBinding binding) {
...
+        showOperationValue(OPERATION_STRING);
+        showResultValue(RESULT_STRING);
+    }
+
+    @Override
+    public void clearLast(String operation) {
+        showOperationValue(operation);
+        showResultValue(RESULT_STRING);
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/calculator/MainActivity.java (hunk 1 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/MainActivity.java:45]: Solved!

File: app/src/main/java/com/example/calculator/MainActivity.java (hunk 3 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/MainActivity.java]: Solved!

File: app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java (hunk 4 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/model/CalculatorModel.java]: Solved!

File: README.md (hunk 5 above)
Comment (DiegoDiazGlobant) [reply README.md:5]: Updated!

File: app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java (hunk 6 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java]: Solved!

File: app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java (hunk 7 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java:33]: Solved!

File: app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java (hunk 8 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/view/CalculatorView.java]: Solved!!

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Pull request merged**